crates/bevy_render/src/render_resource/shader.rs

```diff
@@ -367,17 +367,23 @@ pub static SHADER_IMPORT_PROCESSOR: Lazy<ShaderImportProcessor> =
 pub struct ShaderProcessor {
     ifdef_regex: Regex,
     ifndef_regex: Regex,
+    ifeq_regex: Regex,
+    ifneq_regex: Regex,
     else_regex: Regex,
     endif_regex: Regex,
+    def_regex: Regex,
 }
 
 impl Default for ShaderProcessor {
     fn default() -> Self {
         Self {
             ifdef_regex: Regex::new(r"^\s*#\s*ifdef\s*([\w|\d|_]+)").unwrap(),
             ifndef_regex: Regex::new(r"^\s*#\s*ifndef\s*([\w|\d|_]+)").unwrap(),
+            ifeq_regex: Regex::new(r"^\s*#\s*ifeq\s*([\w|\d|_]+)\s*([\w|\d]+)").unwrap(),
+            ifneq_regex: Regex::new(r"^\s*#\s*ifneq\s*([\w|\d|_]+)\s*([\w|\d]+)").unwrap(),
```

Comment:
How about something like this to make the syntax a bit more familiar?

```suggestion
            ifeq_regex: Regex::new(r"^\s*#\s*if\s*([\w|\d|_]+)\s==\s*([\w|\d]+)).unwrap(),
            ifneq_regex: Regex::new(r"^\s*#\s*if\s*([\w|\d|_]+)\s!=\s*([\w|\d]+)").unwrap(),
```
This would match on `#if TEST == 3` and `#if TEST != 3`